File: src/main/java/com/example/template/template/presentation/sample/form/SampleForm.java
package com.example.template.template.presentation.sample.form;

import jakarta.validation.constraints.NotBlank;
import lombok.Getter;
import org.hibernate.validator.constraints.Length;

/**
 * 追加するDBのサンプル情報をもつFormクラス。
 */
@Getter
public class SampleForm {

  /** コメントの記入者名 */
  @NotBlank(message = "名前は必須です。")
  @Length(max = 10)
  private String name;

  /** コメントの内容 */
  @NotBlank(message = "コメントは必須です。")
  @Length(max = 100)
  private String comment;
}

File: web/src/models/samples/SampleForm.ts
/** コメントを投稿する際のフォームクラス。 */
export default class SampleForm {
  name: string;
  comment: string;

  /**
   * コメントのフォームクラスを生成する。
   *
   * @param name コメントの記入者名
   * @param comment コメントの内容
   */
  constructor(name: string, comment: string) {
    this.name = name;
    this.comment = comment;
  }
}
